### PR TITLE
Improve generated code in capability property accessors

### PR DESCRIPTION
**Pull request by exyi**
This patch should fix number of problems with capability properties:
* it couldn't handle weird property types (fallback),
* it couldn't handle inherited properties correctly
* yet, it was using DotvvmBindableObject.GetValue for many properties, which is unnecesarily slow
* setting a big capability with 1 property set will set all the other properties to their default value - allocating unnecesary memory

Problem was that we was sometimes using direct access to the properties dictionary, and sometimes we were using GetValue (and friends). This patch decides for each property if it's ok to use direct access or not (i.e. it does not override GetValue & is not inherited), and then acts accordingly.

### CODE REVIEW

File: src/Framework/Framework/Binding/DotvvmCapabilityProperty.CodeGeneration.cs [hunk 1]
```diff
@@ -109,22 +109,39 @@ public static (LambdaExpression getter, LambdaExpression setter) CreatePropertyG
...
             {
+                if (property is DotvvmPropertyAlias propertyAlias)
+                    return CreatePropertyAccessors(type, propertyAlias.Aliased);
+
+                // if the property does not override GetValue/SetValue, we'll use
+                // control.properties dictionary directly to avoid virtual method calls
+                var canUseDirectAccess =
+                    !property.IsValueInherited && (
+                    property.GetType() == typeof(DotvvmProperty) ||
+                    property.GetType().GetMethod("GetValue", new [] { typeof(DotvvmBindableObject), typeof(bool) }).DeclaringType == typeof(DotvvmProperty) &&
```
Comment (acizmarik) [inline src/Framework/Framework/Binding/DotvvmCapabilityProperty.CodeGeneration.cs:120]: Maybe change for `nameof(DotvvmBindableObject.GetValue)`

**Review by acizmarik — COMMENTED** [review]
It would be good to check whether this behaviour is tested or not

**Comment by exyi** [comment]
I added the tests (and fixed few found issues :] ), @acizmarik would you have a look again?

File: src/Tests/Runtime/CapabilityPropertyTests.cs [hunk 2]
```diff
@@ -140,6 +223,51 @@ public class TestControl5:
             IObjectWithCapability<TestNestedCapabilityWithPrefix>
         {       
         }
+        // public class TestControl6:
```
Comment (tomasherceg) [inline src/Tests/Runtime/CapabilityPropertyTests.cs:226]: Please remove

**Review by tomasherceg — APPROVED** [review]

**Pull request merged**